### PR TITLE
python3Packages.bottombar: init at 1.0

### PR DESCRIPTION
**Pull request by Conni2461**
###### Description of changes

adding outdating package because we need it to fix python3Packages.nutils which is currently broken

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cheriimoya** [comment]
As mentioned, this package is needed to fix nutils.
So as soon as this pr is merged, I could fix nutils :pray:

File: pkgs/development/python-modules/bottombar/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    rev = "refs/tags/v${version}";
+    hash = "sha256-/3m34HcYmmEf92H3938dYV1Q6k44KaCb9TDx9nDNPnM=";
+  };
+
+  propagatedBuildInputs = [
+    flit-core
+  ];
+
+  # The package only has some "interactive" tests where a user must check for
+  # the correct output and hit enter after every check
```
Comment (mweinelt) [inline pkgs/development/python-modules/bottombar/default.nix:24]: In that case you should set `doCheck = false`.

File: pkgs/development/python-modules/bottombar/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  src = fetchFromGitHub {
+    owner = "evalf";
+    repo = "bottombar";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-/3m34HcYmmEf92H3938dYV1Q6k44KaCb9TDx9nDNPnM=";
+  };
+
+  propagatedBuildInputs = [
+    flit-core
+  ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/bottombar/default.nix:21]:
```suggestion
  nativeBuildInputs = [
    flit-core
  ];
```

That is a build tool.

File: pkgs/development/python-modules/bottombar/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  # The package only has some "interactive" tests where a user must check for
+  # the correct output and hit enter after every check
+  pythonImportsCheck = [
+    "bottombar"
+  ];
+
+  meta = with lib; {
```
Comment (mweinelt) [inline pkgs/development/python-modules/bottombar/default.nix:29]:
```suggestion
  meta = with lib; {
    changelog = "https://github.com/evalf/bottombar/releases/tag/v${version}";
```

File: pkgs/development/python-modules/bottombar/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/development/python-modules/bottombar/default.nix:24]: Don't remove the `pythonImportsCheck`. It's all we have, when t here are no actual tests.

Comment (Conni2461) [reply pkgs/development/python-modules/bottombar/default.nix:24]: i see

**Review by mweinelt — APPROVED** [review]

**Comment by Lassulus** [comment]
Result of `nixpkgs-review pr 240337` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python310Packages.bottombar</li>
    <li>python310Packages.bottombar.dist</li>
    <li>python310Packages.nutils</li>
    <li>python310Packages.nutils.dist</li>
    <li>python311Packages.bottombar</li>
    <li>python311Packages.bottombar.dist</li>
    <li>python311Packages.nutils</li>
    <li>python311Packages.nutils.dist</li>
  </ul>
</details>

**Comment by Lassulus** [comment]
the tests take forever!

**Comment by mweinelt** [comment]
@ofborg build python311.pkgs.nutils

File: pkgs/development/python-modules/nutils/default.nix [hunk 4]
```diff
@@ -44,6 +51,7 @@ buildPythonPackage rec {
 
   meta = with lib; {
     description = "Numerical Utilities for Finite Element Analysis";
+    changelog = "https://github.com/evalf/bottombar/releases/tag/v${version}";
     homepage = "https://www.nutils.org/";
     license = licenses.mit;
     broken = stdenv.hostPlatform.isAarch64;
```
Comment (mweinelt) [inline pkgs/development/python-modules/nutils/default.nix:57]:
```suggestion
```

Builds fine on aarch64-linux/darwin.

Comment (mweinelt) [reply pkgs/development/python-modules/nutils/default.nix:57]: And the changelog linked is wrong.

**Comment by mweinelt** [comment]
@ofborg build python3.pkgs.nutils

File: pkgs/development/python-modules/nutils/default.nix [hunk 5]
```diff
@@ -5,28 +5,35 @@
...
   propagatedBuildInputs = [
     numpy
     treelog
     stringly
+    bottombar
+  ];
+
+  nativeBuildInputs = [
+    flit-core
   ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nutils/default.nix:37]:
```suggestion
  nativeBuildInputs = [
    flit-core
  ];

  propagatedBuildInputs = [
    numpy
    treelog
    stringly
    bottombar
  ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) when naming your commits.

**Comment by cheriimoya** [comment]
> Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) when naming your commits.

do you mean the capital I -> lowercase i?

**Pull request merged**